rule/context-as-argument.go

```diff
@@ -33,20 +60,48 @@ func (r *ContextAsArgumentRule) Name() string {
 }
 
 type lintContextArguments struct {
-	file      *lint.File
-	fileAst   *ast.File
-	onFailure func(lint.Failure)
+	file             *lint.File
+	fileAst          *ast.File
+	allowTypesBefore []string
+	onFailure        func(lint.Failure)
 }
 
 func (w lintContextArguments) Visit(n ast.Node) ast.Visitor {
 	fn, ok := n.(*ast.FuncDecl)
 	if !ok || len(fn.Type.Params.List) <= 1 {
 		return w
 	}
+	allowTypesLookup := make(map[string]struct{}, len(w.allowTypesBefore))
+	for _, v := range w.allowTypesBefore {
+		allowTypesLookup[v] = struct{}{}
```

Comment:
Curious why not a boolean value?

Reply:
I find struct{} is more clear for sets. If it's a string -> bool map, I have to wonder if both true/false in the bool have meaning, or only true (i.e. if it's a set or a map). If the value's a struct{}, I immediately know it's a set of strings.